programs/server/Server.cpp

```diff
@@ -109,11 +113,36 @@ namespace CurrentMetrics
     extern const Metric MaxDDLEntryID;
 }
 
+#if USE_JEMALLOC
+static bool jemallocOptionEnabled(const char *name)
+{
+    bool value;
+    size_t size = sizeof(value);
+
+    if (mallctl(name, reinterpret_cast<void *>(&value), &size, /* newp= */ nullptr, /* newlen= */ 0))
+        throw Poco::SystemException("mallctl() failed");
+
+    return value;
+}
+#else
+static bool jemallocOptionEnabled(const char *) { return 0; }
+#endif
+
 
 int mainEntryClickHouseServer(int argc, char ** argv)
 {
     DB::Server app;
 
+    auto enable_watchdog = [&]()
+    {
+        if (jemallocOptionEnabled("opt.background_thread"))
+        {
+            LOG_ERROR(&app.logger(), "jemalloc.background_thread has been enabled, but it is not compatible with watchdog (due to fork()), watchdog will not be enabled.");
```

Comment:
Not 100% clear, what is the reason?

Reply:
The reason (in jemalloc) is that background_thread, create threads, and fork() will not clone the threads, and that's why it is disable in jemalloc.

This message should not too verbose I guess, it should just underline the problem.

Reply:
So, it's an issue with `jemalloc` - it does not work correctly with fork?

Reply:
Yes (but I guess it can be re-enabled in the child process, manually via mallctl, although not sure that it worth it)